### ISSUE TITLE
Single/Double.Lerp

### ISSUE THREAD

**Issue by oliverbooth** [oliverbooth/X10D#38]
**Type**
```cs
System.Single
System.Double
```

**Extension method signature**
```cs
float Lerp(float, float);
double Lerp(double, double);
```

**Parameters**
| Parameter | Type | Description |
|- |- |- |
|target|`float` / `double`|The target of the interpolation.|
|alpha|`float` / `double`|The interpolation alpha.|

**Description**
Linearly interpolates from the current value to a specified target with a specified alpha.

**Benefits**
Could be implemented using "precise" method as outlined [on Wikipedia](https://en.wikipedia.org/wiki/Linear_interpolation#Programming_language_support).

**Drawbacks**
Deciding a reasonable order of parameters is a task in its own right. i.e. should the format be `value.Lerp(target, alpha)` or `alpha.Lerp(value, target)`?

**(Optional) Implementation example**
```cs
{
    return (1 - alpha) * value + alpha * target;
}
```

**Comment by oliverbooth** [comment]
With thanks to Zenvin, a solution for the mentioned drawback has been considered.

Solution: Offer clearly-named overloads which let the consumer invoke how they wish.
```cs
t.LerpWith(a, b);
a.LerpTo(b, t);
b.LerpFrom(a, t);
```

**Comment by oliverbooth** [comment]
Implemented with 1b811cb18d47a463fdcd35216ab1cb3630d5df73
